client/common/file.c

```diff
@@ -1131,6 +1143,14 @@ BOOL freerdp_client_populate_rdp_file_from_settings(rdpFile* file, const rdpSett
 			return FALSE;
 	}
 
+	if (freerdp_settings_get_bool(settings, FreeRDP_GatewayArmTransport))
+	{
+		freerdp_client_file_string_check_free(file->ResourceProvider);
+		file->ResourceProvider = _strdup("arm");
```

Comment:
maybe use a `static const char resource_provider_arm[] = "arm";` variable ?
also, if the only option is `NULL` and `arm` make the sting `const` to avoid `_studup` and `free` checks?

Reply:
Yes, absolutely. I'll do the chamges later when I have the chance